src/main/java/org/folio/circulation/domain/validation/UniqRequestValidator.java

```diff
@@ -0,0 +1,37 @@
+package org.folio.circulation.domain.validation;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+import org.folio.circulation.domain.Request;
+import org.folio.circulation.domain.RequestAndRelatedRecords;
+import org.folio.circulation.support.HttpFailure;
+import org.folio.circulation.support.Result;
+
+public class UniqRequestValidator {
+
+  private final Function<RequestAndRelatedRecords, ? extends HttpFailure> errorSupplier;
+
+  public UniqRequestValidator(Function<RequestAndRelatedRecords, ? extends HttpFailure> errorSupplier) {
+    this.errorSupplier = errorSupplier;
+  }
+
+  public Result<RequestAndRelatedRecords> refuseWhenRequestIsAlreadyExisted(RequestAndRelatedRecords request) {
```

Comment:
This name suggests to me the check is whether another request exists.

My understanding is that this check is if the same user has already requested the item. 

Maybe a name like `refuseWhenUserHasAlreadyRequestedItem` might describe that better. What do you think?

Reply:
I agree.